### PR TITLE
tf.compat.path_to_str  #25826

### PR DESCRIPTION
**Pull request by shashvatshahi1998**
Added some usage examples #25826

### CODE REVIEW

**Comment by shashvatshahi1998** [comment]
@rthadur  please assign reviewers please

**Comment by shashvatshahi1998** [comment]
please somebody review

**Comment by shashvatshahi1998** [comment]
Please @xiejw  review this pr

**Comment by shashvatshahi1998** [comment]
Please somebody review this PR @rthadur , @tomhennigan , @mihaimaruseac

File: tensorflow/python/util/compat.py
Comment (mihaimaruseac) [inline tensorflow/python/util/compat.py]: Add an empty line above `Examples:` please

File: tensorflow/python/util/compat.py
Comment (mihaimaruseac) [inline tensorflow/python/util/compat.py]: Shouldn't it be `XYZ\\tensorflow` (double backslash)?

Can you actually run it and see what the output should be when backslashes and slashes are mixed?

File: tensorflow/python/util/compat.py
Comment (mihaimaruseac) [inline tensorflow/python/util/compat.py]: Same as above

**Review by mihaimaruseac — COMMENTED** [review]
Looks good, minor nits and a request to actually run the examples

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac  I have done the changes requested by you, now can you please review it

**Comment by shashvatshahi1998** [comment]
@rthadur , @mihaimaruseac please somebody review it.

**Comment by shashvatshahi1998** [comment]
Please @mihaimaruseac  review it now.

File: tensorflow/python/util/compat.py
Comment (mihaimaruseac) [inline tensorflow/python/util/compat.py]: Trailing whitespace

**Review by mihaimaruseac — COMMENTED** [review]
Thank you for the updates, there is a trailing whitespace which you should remove and then all is good.

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac I am not able to remove that white trailing space, I tried for 10 times atleast how to remove that there is nothing to be removed , please help.

**Comment by mihaimaruseac** [comment]
In your editor, you go to that line and press delete/backspace a few times.

Most editors can be configured to remove trailing whitespace on save. What editor are you using?

**Comment by shashvatshahi1998** [comment]
I am using vim editor

**Comment by shashvatshahi1998** [comment]
Done with changes @mihaimaruseac , now please approve it and merge it.

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac,@rthadur why there is ubuntu sanity check , please somebody guide me.

**Comment by shashvatshahi1998** [comment]
Please, @mihaimaruseac, @rthadur help me with this ubuntu sanity check, I'm not getting the reason for the error.

**Comment by mihaimaruseac** [comment]
Click on details then on log

```
1. do_pylint PYTHON2: Python 2 pylint
  FAIL
2. do_pylint PYTHON3: Python 3 pylint
  FAIL
```

Basically you're not following PEP8 style guide so code cannot be merged. Please fix.

**Comment by shashvatshahi1998** [comment]
What's wrong in my style I am not able to understand.

**Comment by shashvatshahi1998** [comment]
I have tried to change a little bit can you please tell me what can I do further.

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac  please review it once and tell me further changes.

**Comment by mihaimaruseac** [comment]
You didn't fix the error, instead changed irrelevant lines.

If you look at the [error log](https://source.cloud.google.com/results/invocations/7a6a3039-f030-40c3-a3d0-20605ad514e2/log) it says right there

```
FAIL: Found 1 non-whitelited pylint errors:
tensorflow/python/util/compat.py:123: [C0301(line-too-long), ] Line too long (89/80)
```

You can run the linter locally too, see https://pypi.org/project/pep8/

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac  I have reduced the lengths of all the lines(<80) according to pylint norms, so now can you please review it.

File: tensorflow/python/util/compat.py
Comment (mihaimaruseac) [inline tensorflow/python/util/compat.py]: Don't change this line. I know it is already too long but first phrase in docstring should always be on a single line and followed by empty space (please run linter locally and check the warnings issued on the lines you're changing).

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac I have done the changes requested by you now you can review it.

**Review by mihaimaruseac — APPROVED** [review]
Thank you. I think it will get merged on Monday though

**Comment by shashvatshahi1998** [comment]
@rthadur please apply kokoro force run

**Comment by shashvatshahi1998** [comment]
Please @rthadur apply all checks

**Comment by mihaimaruseac** [comment]
It's weekend, people are not going to work on this until Monday. Please wait.

**Comment by mihaimaruseac** [comment]
The Windows Bazel GPU check: don't worry about it, it's not your PR's fault. But until MondayTuesday there likely won't be a fix to it.

**Comment by shashvatshahi1998** [comment]
Please somebody check the internal CI build please.

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac please now its more than 2 days please assign someone to check import/copybara

**Comment by mihaimaruseac** [comment]
It's Monday morning, people have not yet come to the office. When @rthadur comes in he will apply the copybara checks.

There is one issue though, which you should fix before: since you include commits changing other files, copybara sees a change in `api_def_Bitcast.pbtxt` which shouldn't be (but it is included because of d3c1f4a).

You could try rebasing/squashing these commits to only have one single commit doing the exact change this PR does. Not fixing this will make merging the PR slower

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac why merging is still blocked when all required checks are done?

**Comment by mihaimaruseac** [comment]
Because `￼Windows Bazel GPU — Internal CI build failed`

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac  what can I do for Windows Bazel GPU test.

**Comment by mihaimaruseac** [comment]
Just wait.

**Comment by mihaimaruseac** [comment]
That test fails in several PRs, just wait for it to be fixed and then the PR will be merged automatically. Unless mentioned again, you don't have anything else to do for this PR (except maybe clean it up, squash/remove old commits)

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac what happened to it now 4 tests are failing why?

**Comment by mihaimaruseac** [comment]
Other tests failed. They were run because your PR touched other files so they were marked as required to run before merging. Please try to have clean PRs in the future.

You don't have to do anything here though, this will get merged in a few minutes

**Comment by mihaimaruseac** [comment]
See 66d7a531f as the commit merging this.

**Comment by shashvatshahi1998** [comment]
@mihaimaruseac sure, I will try to submit PRs with no unwanted commits.

**Pull request closed without merging**
